tests/test_repository_pypi.py

```diff
@@ -264,6 +287,64 @@ def _get_project(self, ireq):
     assert actual_hashes == expected_hashes
 
 
+def test_get_hashes_from_mixed(pip_conf, from_line, tmpdir):
```

Comment:
Could you please also add an integrational test to [test_cli_compile.py](https://github.com/jazzband/pip-tools/blob/master/tests/test_cli_compile.py)?